library/src/cache/patch_manager.rs

```diff
@@ -836,9 +911,13 @@ mod next_boot_patch_tests {
         manager.record_boot_start_for_patch(1)?;
         manager.record_boot_success()?;
 
+        // Add a second patch and record that we started booting it, but not that it succeeded.
         manager.add_patch_for_test(&temp_dir, 2)?;
         manager.record_boot_start_for_patch(2)?;
 
+        // Simulate a restart.
```

Comment:
// Simulate that the app is being started fresh (e.g. from a crash)
// not bothering to re-load the state from disk.

Reply:
Same as above